ngraph_bridge/ngraph_utils.cc

```diff
@@ -463,6 +475,201 @@ bool IsProcessedByNgraphPass(Graph* g) {
   return false;
 }
 
+Status DumpNGraph(int file_idx, tensorflow::GraphDef* graph_def,
+                  std::set<std::string> nodes) {
+  const char* path = std::getenv("NGRAPH_TF_TB_LOGDIR");
+
+  if (path == nullptr) {
+    return Status::OK();
+  }
+
+  std::string str_path(path);
+
+  if (str_path.back() != '/') {
+    str_path += "/";
+  }
+
+  struct stat buffer;
+  if (stat(str_path.c_str(), &buffer) != 0)  // path doesn't exist
+  {
+    mkdir(str_path.c_str(), 0777);
+  }
+
+  std::string dname = GetSessionName(file_idx, nodes);
+  str_path += (dname.insert(0, "ngraph") + "/");
+
+  if (stat(str_path.c_str(), &buffer) != 0)  // path doesn't exist
+  {
+    mkdir(str_path.c_str(), 0777);
```

Comment:
minor:
is 0777 the appropriate mode? Can we have a less permissive mode?

same comment for other 0777s

Reply:
I think I see this code snippet repeated. Can we move it to a function?
```
if (stat(str_path.c_str(), &buffer) != 0)  // path doesn't exist
  {
    mkdir(str_path.c_str(), 0777);
}
```

Reply:
will move to a separate function

kept 0777 because i wanted user to not worry about any permission issues for their experiments. if change is needed, what permissions do we want for user/group/other? rwxrwx---? or rwx------?

Reply:
doing 0750. user can rwx, group can rw, other can't do anything